### ISSUE TITLE
TileSprite create => destroy => create sequence breaks Phaser

### ISSUE THREAD

**Issue by eabatalov** [photonstorm/phaser#1753]
Hello. Here is a minimal reproducer:
```javascript
    var xxx = game.make.tileSprite(0, 0, 256, 256, 'player');
    game.world.add(xxx);
    
    setTimeout(function() {
        xxx.destroy();
        setTimeout(function() {
            xxx = game.make.tileSprite(0, 0, 256, 256, 'player');
            game.world.add(xxx);
        }.bind(this), 5000);
    }.bind(this), 5000);
```

I've pasted it to 'create' tab here https://phaser.io/sandbox/edit/bBXJeYyG run the example and wait 10 seconds to reproduce the bug.

The delays are needed to reproduce the bug. I guess rendering should be done at least once. This bug makes TileSprites unusable for me. It wasn't observed in Phaser 2.2.0

This bug looks hard so I can't help in fixing.

**Comment by vulvulune** [comment]
Same problem if you have a tileSprite in a state, you load another state, and reload the first state again. When the tileSprite is added, there is a "Type mismatch error" on the line
"this.__tilePattern = context.createPattern(this.tilingTexture.baseTexture.source, 'repeat');" (PIXI.TilingSprite)

This problems occurs only with the version 2.3.0 of Phaser in canvas mode.

I will try to find which commits add this issue but I think it's a PIXI bug... :-(

**Comment by vulvulune** [comment]
With the last version (dev branch), it works!!!

**Comment by eabatalov** [comment]
Cool. Hope we have the fix in the next stable Phaser update. Don't known when it is planned to be released.

**Comment by xingxp** [comment]
while waiting the new 2.3.1, should i use 2.2 instead of current version?
this bug really boring.

**Comment by vulvulune** [comment]
Yes, I think you should use the 2.2 or compile a dev version.

**Comment by netgfx** [comment]
True the dev version fixes this issue.

**Comment by photonstorm** [comment]
Closing as this is now fixed in dev and 2.4 will be out very very soon (this week I hope)
